### ISSUE TITLE
Install missing tools automatically on the fly

### ISSUE THREAD

**Issue by hohwille** [devonfw/ide#281]
As a user, I want `devon-ide` to install missing tools on the fly so that I do not get errors for missing tools and have a better UX.

This requirement originates from #265 
If we call `devon «comandlet» «args»` where `«args»` is not setup and the tool corresponding to `«commandlet»` is not installed or that tools requires another tool (e.g. `eclipse` -> `Java`) then currently we get errors.
This issue wants to change this such that the installation is checked transitive so all required tools are properly installed. To prevent accidental installations there should be a (single) user confirmation before such installation starts:
```
devon eclipse
The tool eclipse is not installed in your current devon-ide. Do you want to install it now?
(yes/no): yes
...installing eclipse
...installing Java
...starting eclipse
```
In order to align this feature with #177 we should also improve our output/logging to reduce unimportant stuff. Therefore we do not have to see `checking if Java installed, blabla` every time we launch eclipse.

**Comment by hohwille** [comment]
Fixed with PR #324

**Comment by hohwille** [comment]
FYI: The implication is now that any manual changes in software are immediately reverted. So to fix a release that was not working with Java11 I simply changed `software/java` to be a symlink to a Java8 installation. Then invoking a build with `devon` already reverted this change. So I had to change `JAVA_VERSION` in my `devon.properties` to get Java8. Just a feedback to gain awareness for the implication of the change. We should add this to the documentation... Would be awesome if someone has a few minutes to do that...
